src/tests/intg/test_files_provider.py

```diff
@@ -0,0 +1,689 @@
+#
+# SSSD files domain tests
+#
+# Copyright (c) 2016 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 only
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import stat
+import time
+import config
+import signal
+import subprocess
+import pytest
+import tempfile
+import shutil
+import pwd
+import grp
+
+import ent
+import sssd_id
+from sssd_nss import NssReturnCode
+from sssd_passwd import call_sssd_getpwnam, call_sssd_enumeration
+from sssd_group import call_sssd_getgrnam
+from files_ops import passwd_ops_setup, group_ops_setup
+from util import unindent
+
+# FIXME - remove before pushing
+from util import run_shell
```

Comment:
I use git stash for `run_shell` and debugging purposes. Then  will never forget to remove it.

BTW here is a diff  for fixing imports
```
@@ -22,11 +22,9 @@ import time
 import config
 import signal
 import subprocess
-import pytest
-import tempfile
-import shutil
 import pwd
 import grp
+import pytest
 
 import ent
 import sssd_id
@@ -36,8 +34,6 @@ from sssd_group import call_sssd_getgrnam
 from files_ops import passwd_ops_setup, group_ops_setup
 from util import unindent
 
-# FIXME - remove before pushing
-from util import run_shell
```